config.json

```diff
@@ -1560,9 +1560,10 @@
       {
         "groupId": "com.google.android.material",
         "artifactId": "compose-theme-adapter",
-        "version": "1.1.16",
-        "nugetVersion": "1.1.16",
+        "version": "1.1.17",
+        "nugetVersion": "1.1.17",
         "nugetId": "Xamarin.Google.Android.Material.Compose.Theme.Adapter",
+        "excludedRuntimeDependencies": "androidx.compose.runtime.runtime,androidx.compose.material.material",
```

Comment:
What are these exclusions for?

Reply:
google has more and more stable packages that depend on previews

either our scripts will suffer or I will suffer with binding previews

```
Dependency errors : 2
1
	System.Exception: 
Artifact
	com.google.android.material.compose-theme-adapter:1.1.17
has unknown 'Runtime' dependency 
	androidx.compose.runtime.runtime:1.3.0-beta01
Either fulfill or exclude this dependency.

2
	System.Exception: 
Artifact
	com.google.android.material.compose-theme-adapter:1.1.17
has unknown 'Runtime' dependency 
	androidx.compose.material.material:1.3.0-beta01
Either fulfill or exclude this dependency.
```

Reply:
I think we need to just leave this package at the existing version until there is a newer version that does not require preview packages.  Excluding the dependencies will create a broken package that does not function for users.

We have two options for remaining on the current version:
- Manually revert it each week until a version that only depends on stable packages is released.
- Set it as `"frozen": true` and the script will not update the version any more.

Reply:
> We have two options for remaining on the current version:
> 
> * Manually revert it each week until a version that only depends on stable packages is released.

Right. That is the reason I mentioned that "scripts will suffer" (must be modified) or I have to manually check every week or bind the preview.

> * Set it as `"frozen": true` and the script will not update the version any more.

I will set it to `frozen`, hoping will not forget to unfreeze it soon.

```
<joke-sarcasm>
BTW, for the sake of inclusiveness can we change `frozen` to something else? `frozen` is scary word in Europe right now and not politically correct.
<j/oke-sarcasm>
```

Reply:
for the record:

Freezing 2 artifacts leads to:

```
Dependency errors : 17
1
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.runtime.runtime:1.2.1
to satisfy dependency of: 
			androidx.compose.foundation.foundation:1.2.1

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.runtime",
        "artifactId": "runtime",
        "version": "1.2.1",
        "nugetVersion": "CHECK PREFIX 1.2.1",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


2
	System.Exception: 
Artifact
	androidx.compose.foundation.foundation-layout:1.2.1
has unknown 'Runtime' dependency 
	androidx.compose.runtime.runtime:1.2.1
Either fulfill or exclude this dependency.

3
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.material.material-icons-core:[1.2.0]
to satisfy dependency of: 
			androidx.compose.material.material:1.2.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.material",
        "artifactId": "material-icons-core",
        "version": "[1.2.0]",
        "nugetVersion": "CHECK PREFIX [1.2.0]",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


4
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.material.material-ripple:[1.2.0]
to satisfy dependency of: 
			androidx.compose.material.material:1.2.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.material",
        "artifactId": "material-ripple",
        "version": "[1.2.0]",
        "nugetVersion": "CHECK PREFIX [1.2.0]",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


5
	System.Exception: 
Artifact
	androidx.compose.material.material-icons-extended:1.2.1
has unknown 'Runtime' dependency 
	androidx.compose.runtime.runtime:1.2.1
Either fulfill or exclude this dependency.

6
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.runtime.runtime:[1.2.1]
to satisfy dependency of: 
			androidx.compose.runtime.runtime-livedata:1.2.1

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.runtime",
        "artifactId": "runtime",
        "version": "[1.2.1]",
        "nugetVersion": "CHECK PREFIX [1.2.1]",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


7
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.runtime.runtime:[1.2.1]
to satisfy dependency of: 
			androidx.compose.runtime.runtime-rxjava2:1.2.1

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.runtime",
        "artifactId": "runtime",
        "version": "[1.2.1]",
        "nugetVersion": "CHECK PREFIX [1.2.1]",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


8
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.runtime.runtime:[1.2.1]
to satisfy dependency of: 
			androidx.compose.runtime.runtime-saveable:1.2.1

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.runtime",
        "artifactId": "runtime",
        "version": "[1.2.1]",
        "nugetVersion": "CHECK PREFIX [1.2.1]",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


9
	System.Exception: 
Artifact
	androidx.compose.ui.ui:1.2.1
has unknown 'Runtime' dependency 
	androidx.compose.runtime.runtime:1.2.1
Either fulfill or exclude this dependency.

10
	System.Exception: 
Artifact
	androidx.compose.ui.ui-text:1.2.1
has unknown 'Runtime' dependency 
	androidx.compose.runtime.runtime:1.2.1
Either fulfill or exclude this dependency.

11
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.runtime.runtime:1.2.1
to satisfy dependency of: 
			androidx.compose.ui.ui-tooling-data:1.2.1

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.runtime",
        "artifactId": "runtime",
        "version": "1.2.1",
        "nugetVersion": "CHECK PREFIX 1.2.1",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


12
	System.Exception: 
Artifact
	com.google.accompanist.accompanist-appcompat-theme:0.25.1
has unknown 'Runtime' dependency 
	androidx.compose.material.material:1.2.1
Either fulfill or exclude this dependency.

13
	System.Exception: 
No matching artifact config found for: 
			androidx.compose.material.material:1.2.1
to satisfy dependency of: 
			com.google.accompanist.accompanist-pager-indicators:0.25.1

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.compose.material",
        "artifactId": "material",
        "version": "1.2.1",
        "nugetVersion": "CHECK PREFIX 1.2.1",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


14
	System.Exception: 
Artifact
	com.google.accompanist.accompanist-placeholder-material:0.25.1
has unknown 'Runtime' dependency 
	androidx.compose.material.material:1.2.1
Either fulfill or exclude this dependency.

15
	System.Exception: 
Artifact
	com.google.accompanist.accompanist-swiperefresh:0.25.1
has unknown 'Runtime' dependency 
	androidx.compose.material.material:1.2.1
Either fulfill or exclude this dependency.

16
	System.Exception: 
Artifact
	com.google.android.material.compose-theme-adapter:1.1.17
has unknown 'Runtime' dependency 
	androidx.compose.runtime.runtime:1.3.0-beta01
Either fulfill or exclude this dependency.

17
	System.Exception: 
Artifact
	com.google.android.material.compose-theme-adapter:1.1.17
has unknown 'Runtime' dependency 
	androidx.compose.material.material:1.3.0-beta01
Either fulfill or exclude this dependency.
```

Reply:
> I do not see those errors locally. I pushed the change to the branch to see if CI has them.

Really? I am fighting dependency hell.

Now on 1 error:

```
Dependency errors : 1
1
	System.Exception: 
Artifact
	com.google.android.material.compose-theme-adapter:1.1.14
has unknown 'Runtime' dependency 
	com.google.android.material.material:1.7.0-alpha02
Either fulfill or exclude this dependency.
```

And we have published 1.1.16.

Could it be that google added runtime dependencies retroactively?

Reply:
I think the commit I did was correct.  It looks like for `com.google.android.material.compose-theme-adapter`, version `1.1.16` does not rely on preview dependencies, but versions `1.1.14`, `1.1.15`, and `1.1.17` do.  Hence freezing it at `1.1.16`.

I do not believe there is any need to freeze any other artifacts.